src/log.ts

```diff
@@ -136,36 +136,35 @@ export class Log {
         requestInit.signal = controller.signal as AbortSignal;
         requestInit.method = 'GET';
 
-        await fetch(requestURL.toString(), requestInit)
-            .then((response) => {
-                const status = response.status;
-                if (status === 200) {
-                    // TODO: the follow search param still has the stream close prematurely based on my testing
-                    response.body.pipe(stream);
-                } else if (status === 500) {
-                    const v1status = response.body as V1Status;
-                    const v1code = v1status.code;
-                    const v1message = v1status.message;
-                    if (v1code !== undefined && v1message !== undefined) {
-                        throw new ApiException<undefined | V1Status>(
-                            v1code,
-                            v1message,
-                            v1status,
-                            normalizeResponseHeaders(response),
-                        );
-                    }
-                } else {
-                    throw new ApiException<undefined>(
-                        status,
-                        'Error occurred in log request',
-                        undefined,
+        try {
+            const response = await fetch(requestURL.toString(), requestInit);
+            const status = response.status;
+            if (status === 200) {
+                // TODO: the follow search param still has the stream close prematurely based on my testing
+                response.body.pipe(stream);
+            } else if (status === 500) {
+                const v1status = response.body as V1Status;
+                const v1code = v1status.code;
+                const v1message = v1status.message;
+                if (v1code !== undefined && v1message !== undefined) {
+                    throw new ApiException<undefined | V1Status>(
+                        v1code,
+                        v1message,
+                        v1status,
                         normalizeResponseHeaders(response),
                     );
                 }
-            })
-            .catch((err) => {
-                throw new ApiException<undefined>(err, 'Error occurred in log request', undefined, err);
-            });
+            } else {
+                throw new ApiException<undefined>(
+                    status,
+                    'Error occurred in log request',
+                    undefined,
+                    normalizeResponseHeaders(response),
+                );
+            }
+        } catch (err: any) {
+            throw new ApiException<undefined>(err, 'Error occurred in log request', undefined, err);
```

Comment:
Note - this line is ported from the previous `catch()` call, but I don't think passing `err` as the first parameter is actually correct. `catch()` typed `err` as `any`. When I changed to a `try...catch`, that typing was not there and TypeScript complained that the first argument should be a numeric code (probably similar to how `src/metrics.ts` does it).

Reply:
Yeah, I'm on board with switching this to match metric.ts